src/fprime/util/cli.py

```diff
@@ -1,170 +1,52 @@
+        # runners is a Dict[str, Callable] of {command_name: handler_functions} pairs
+        return runners[parsed.command](
+            build, parsed, cmake_args, make_args, getattr(parsed, "pass_through", [])
         )
-    print("[INFO] File(s) associated with hash 0x{:x}".format(parsed.hash))
-    for line in lines:
-        print("   ", line, end="")
-
-
-def template(
-    build: Build, parsed: argparse.Namespace, _: Dict[str, str], __: Dict[str, str], ___
-):
-    """Processes new command
-
-    Args:
-        build: build used to inform new component and new port calls
-        parsed: parsed arguments
-        _: unused cmake arguments
-        __: unused make arguments
-        ___: unused pass through arguments
-    """
-    if parsed.component and not parsed.port:
-        return new_component(build.deployment, parsed.platform, parsed.verbose, build)
-    if parsed.port and not parsed.component:
-        return new_port(build.deployment, build)
-    print("[ERROR] Use --component or --port, not both.", file=sys.stderr)
+    except Exception as exc:
+        print(f"[ERROR] {exc}", file=sys.stderr)
+    return 1
 
 
-def run_code_format(
-    build: Build,
-    parsed: argparse.Namespace,
-    _: Dict[str, str],
-    __: Dict[str, str],
-    ___: List[str],
-):
-    """Runs code formatting using clang-format
-
-    Args:
-        build: used to retrieve .clang-format file
-        parsed: parsed input arguments
-        __: unused cmake_args
-        ___: unused make_args
-        ____: unused pass-through arguments
-    """
-    options = {
-        "verbose": not parsed.quiet,
-        "backup": not parsed.no_backup,
-        "validate_extensions": not parsed.force,
-    }
-    clang_formatter = ClangFormatter(
-        "clang-format",
-        build.settings.get("framework_path", Path(".")) / ".clang-format",
-        options,
-    )
-    if not clang_formatter.is_supported():
-        print(
-            f"[ERROR] Cannot find executable: {clang_formatter.executable}. Unable to run formatting.",
-            file=sys.stderr,
-        )
-        return 1
-    # Allow requested file extensions
-    for file_ext in parsed.allow_extension:
-        clang_formatter.allow_extension(file_ext)
-    # Stage all files that are passed through stdin, if requested
-    if parsed.stdin:
-        for filename in sys.stdin.read().split():
-            clang_formatter.stage_file(Path(filename))
-    # Stage all files that are passed through --files
-    for filename in parsed.files:
-        clang_formatter.stage_file(Path(filename))
-    return clang_formatter.execute(build, parsed.path, ({}, parsed.pass_through))
+def skip_build_loading(parsed):
+    """Determines if the build load step should be skipped. Commands that do not require a build object
+    should manually be added here by the developer."""
+    if parsed.command == "new" and parsed.deployment:
+        return True
+    return False
```

Comment:
Aren't there more items that skip the load build?  E.g. info?

Reply:
Currently, all other commands do need a build object. For `info`, that's [here](https://github.com/fprime-community/fprime-tools/blob/9e0191425206de2b35c7a3dde3ae60327414c685/src/fprime/util/cli.py#L40)
Although you're making a good point, with that new code structure there is probably cleaner ways to handle those edge cases and add them here (e.g. `format` only needs a project). I'll take note and look into how this can be dealt with later, but I think we can move forward with this for now.